mkdocs/nav.py

```diff
@@ -205,15 +205,19 @@ def set_canonical_url(self, base):
 
     def set_edit_url(self, repo_url, edit_uri):
         if not repo_url.endswith('/'):
-            repo_url += '/'
+            # Skip when using query or fragment in edit_uri
+            if not edit_uri.startswith('?') and not edit_uri.startswith('#'):
+                repo_url += '/'
         if not edit_uri:
             self.edit_url = repo_url
         else:
             if not edit_uri.endswith('/'):
                 edit_uri += '/'
+            # Normalize URL from Windows path '\\' -> '/'
+            input_path_url = self.input_path.replace('\\', '/')
```

Comment:
Because it uses `pathname2url` which will turn the '?' and '#' to their encoded versions and generate wrong URLs.